src/app/index/metadata.rs

```diff
@@ -90,7 +74,41 @@ fn read_id3(path: &Path) -> Result<SongTags> {
 		track_number,
 		year,
 		has_artwork,
-	})
+	}
+}
+
+fn read_mp3(path: &Path) -> Result<SongTags> {
+	let tag = id3::Tag::read_from_path(&path).or_else(|error| {
+		if let Some(tag) = error.partial_tag {
+			Ok(tag)
+		} else {
+			Err(error)
+		}
+	})?;
+
+	let duration = {
+		mp3_duration::from_path(&path)
+			.map(|d| d.as_secs() as u32)
+			.ok()
+	};
+
+	let mut song_tags = read_id3(&tag);
+
```

Comment:
Nitpick: can you remove this blank line? I tend to use them to separate logically independent blocks (although it it isn't a perfect science and I can't blame you for not reading my mind).

Same note for the blank lines inside `read_mp3` and `read_wave` below.

Reply:
No problem.